app/controllers/nodes_controller.rb

```diff
@@ -16,6 +18,18 @@ def show
     @minion = Minion.find(params[:id])
   end
 
+  def bootstrap
+    salt = Pharos::Salt.new
+    available_roles = %i(master)
+    salt.minions.keys.each do |minion|
+      Minion.transaction do
+        Minion.find_or_create_by!(hostname: minion).assign_role(available_roles)
```

Comment:
The master role will only be applied to one node. After that role is assigned (it's given priority for assignment), we will just assign minion roles.